src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -164,18 +164,7 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
       ignoreVersionMismatch: config.ignoreVersionMismatch,
       esVersionCheckInterval: config.healthCheckDelay.asMilliseconds(),
       kibanaVersion: this.kibanaVersion,
-    }).pipe(publishReplay(1));
-
-    this.subscriptions.esNodesCompatibility = (esNodesCompatibility$ as ConnectableObservable<
-      unknown
-    >).connect();
-
-    // TODO: Move to Status Service https://github.com/elastic/kibana/issues/41983
-    esNodesCompatibility$.subscribe(({ isCompatible, message }) => {
-      if (!isCompatible && message) {
-        this.log.error(message);
-      }
-    });
+    }).pipe(shareReplay({ refCount: true, bufferSize: 1 }));
```

Comment:
Do you use `bufferSize: 1` to reduce memory footprint?

Reply:
The buffer size is the size of the replay subject that `shareReplay` uses internally. As long as there are subscribers, any new subscribers will get the last `1` values. So as long as we're polling, any new subscribers will immediately get the latest version check results.

When there are no subscribers, refCount drops to 0 and the internal replay subject will unsubscribe from `pollEsNodesVersion$`. When there's no subscriptions and then we get one new subscriber, that subscriber won't get any buffered/replayed version checks since these could be very old. Only once `pollEsNodesVersion$` emits a value again will the first subscriber get a result.